### PR TITLE
Remove the model argument from Lite's `optimizer_step` via structural typing

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

See title

### Does your PR introduce any breaking changes? If yes, please list them.

None

cc @borda @justusschock @awaelchli @rohitgr7 @carmocca

### CODE REVIEW

File: src/lightning_lite/utilities/types.py [hunk 1]
```diff
@@ -77,3 +77,11 @@ def __init__(
 
     def step(self, metrics: Union[float, int, Tensor], epoch: Optional[int] = None) -> None:
         ...
+
+
+@runtime_checkable
+class Steppable(Protocol):
+    """To structurally type ``optimizer.step()``"""
+
+    def step(self, closure: Optional[Callable[[], float]] = ...) -> Optional[float]:
```
Comment (awaelchli) [inline src/lightning_lite/utilities/types.py:86]: what about the kwargs? we need to be able to pass them down

**Comment by awaelchli** [comment]
@carmocca I took care of updating the tests

**Review by awaelchli — APPROVED** [review]

**Comment by carmocca** [comment]
> @carmocca I took care of updating the tests

Thanks! I moved the Steppable assertion into the test so we don't need to mock the engine during the "check step called" tests

**Review by justusschock — APPROVED** [review]

File: src/lightning_lite/utilities/types.py (hunk 1 above)
Comment (justusschock) [reply src/lightning_lite/utilities/types.py:87]: could we do this with overloads so that still one of the "options" would match the optimizer?

Comment (awaelchli) [reply src/lightning_lite/utilities/types.py:87]: @carmocca ok, so we would need to put the type ignore in the usages where we pass kwargs i guess

Comment (awaelchli) [reply src/lightning_lite/utilities/types.py:87]: Feel free to resolve my part of this conversation or maybe try what Justus said.

**Pull request merged**